Bottom: 52e399a0e99c45ef78ee5633f785c2b9173ae8ec
Top:    4f77acc51843b31c78728752911e198c56d218e2
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 14:30:16 +0000

Calculate epsilon estimate for debugging


---

diff --git a/src/polygon.c b/src/polygon.c
index 033928c..19947eb 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -123,6 +123,7 @@ void
 polygon_init (void)
 {
   struct rlimit limit;
+  double eps;
 
   double cos_ang = cos (2.0 * M_PI / POLY_CIRC_SEGS_F);
   double sin_ang = sin (2.0 * M_PI / POLY_CIRC_SEGS_F);
@@ -137,6 +138,12 @@ polygon_init (void)
   getrlimit (RLIMIT_AS, &limit);
   limit.rlim_cur = MIN (limit.rlim_cur, 2000 * 1024 * 1024 /* 2 GiB limit to virtual memory size */);
   setrlimit (RLIMIT_AS, &limit);
+
+  /* Estimate eps on this machine */
+  eps = 1.;
+  do eps = 0.5 * eps; while (eps + 1.0 > 1.0);
+
+  fprintf (stderr, "Epsilon calculated at %.17e\n", eps);
 }
 
 Cardinal
